astropy/wcs/utils.py

```diff
@@ -149,26 +149,30 @@ def wcs_to_celestial_frame(wcs):
                      "the specified WCS object")
 
 
-def celestial_pixel_scale(inwcs, allow_nonsquare=False):
+def celestial_pixel_scale(inwcs, which='combine', allow_nonsquare=False):
     """
-    For a WCS, if the pixels are square, return the pixel scale in the spatial
-    dimensions
+    For a WCS, return the pixel scale in the spatial dimensions
 
     Parameters
     ----------
     inwcs: `astropy.wcs.WCS`
         The world coordinate system object
+    which: one of 'combine', 'lon', 'lat', or 'both'
+        The dimension to calculate the pixel scale along. If 'both', a tuple
+        will be returned. If 'combine' with nonsquare pixels, a geometric mean
+        is taken unless `allow_nonsquare=False`
     allow_nonsquare : bool
-        Return the average of the X and Y scales if True
+        Return the geometric average of the X and Y scales if True
 
     Returns
     -------
-    scale : float
-        The square pixel scale
+    scale : `Quantity` or tuple of `Quantity`
```

Comment:
No need for the backticks here -- it causes formatting errors in the documentation.